test/test-buildSrcSet.mjs

```diff
@@ -2,489 +2,348 @@ import md5 from 'md5';
 import assert from 'assert';
 import ImgixClient from '../src/main.mjs';
 
+function assertWidthsIncreaseByTolerance(srcset, tolerance) {
+  const srcsetWidths = srcset.split(',').map((u) => {
+    const tail = u.split(' ')[1];
+    const width = tail.slice(0, -1);
+    return Number.parseFloat(width);
+  });
+
+  // Make two equal sized arrays one for the numerators, e.g.
+  // [x1, x2, ..., xN] and another for the denominators, e.g.
+  // [x0, x1,..., x(N-1)].
+  const numerators = srcsetWidths.slice(1);
+  const denominators = srcsetWidths.slice(0, -1);
+
+  // Zip the numerator/denominator pairs.
+  const pairs = numerators.map((n, i) => {
+    return [n, denominators[i]];
+  });
+
+  // Be as tolerant as we can.
+  const tolerancePlus = tolerance + 0.004;
```

Comment:
Say we're trying to ensure that each subsequent width is 40% larger than the last. We're doing floating point arithmetic so we have to be careful when doing comparisons. In the test where we're ensuring 40% increases we get these ratios:

```
0.40145...
0.40104...
```

If these were convented to percentages, i.e. `40.1%`, and then rounded they'd still be `40%`. The last number to round down to 40% would be `40.4%` (whereas `Math.round(40.5) == 41` which would mean we're checking for 41% instead of 40%).

We could loosen it a bit more if we want to. I just noticed that in our test where we were checking for `0.40` increases the bound we were checking against was actually `2.4`. This might have happened because 1.40104 is not less than 1.40 but is less than 2.4.